bgpd/bgp_vpn.c

```diff
@@ -223,7 +227,7 @@ int show_adj_route_vpn(struct vty *vty, struct peer *peer,
 			output_count++;
 		}
 
-		if (use_json)
+		if (use_json && json_routes)
```

Comment:
could it be that `json_routes` would be NULL here because I see it's always created?:
```
				if (use_json) {
					json_routes = json_object_new_object();
```
just curious: Isn't enough to reset it as you did before? `json_routes = NULL;`?

Reply:
Hi Donatas, nice question! json_routes is created deep within the second for loop:
> for (rm = bgp_table_top(table); rm; rm = bgp_route_next(rm)) {

So, 
1) if there is a chance that this second for loop is never executed, or, 
2) doesnt get the chance to reach the point where json_routes is created, due to these  "continue" statements.
>			if (bgp_node_get_bgp_path_info(rm) == NULL)
>				continue;
>
>			if (!attr)
>				continue;
we may reach this line json_object_object_add(json_adv, rd_str, json_routes) while json_routes is still NULL.